src/ContentScript/index.ts

```diff
@@ -2,64 +2,80 @@
 import { browser } from 'webextension-polyfill-ts';
-// import { debounce } from 'lodash';
 import contentConnector from './event-center/content-connector';
 
-let hidden: any;
-let visibilityChange: any;
-if (typeof document.hidden !== 'undefined') {
-  // Opera 12.10 and Firefox 18 and later support
-  hidden = 'hidden';
-  visibilityChange = 'visibilitychange';
-} else if (typeof document.hidden !== 'undefined') {
-  hidden = 'msHidden';
-  visibilityChange = 'msvisibilitychange';
-  // @ts-ignore
-} else if (typeof document.webkitHidden !== 'undefined') {
-  hidden = 'webkitHidden';
-  visibilityChange = 'webkitvisibilitychange';
-}
-
 class PageInjector {
+  recursiveQueue: number[] = [];
+
+  recursiveNumber = 0;
+
   constructor() {
     contentConnector.init();
-    const injectAll = this.injectInpage.bind(this);
-    // inject on load
-    window.onload = () => {
-      // @ts-ignore
-      if (!document[hidden]) {
-        injectAll();
-      }
-    };
-    // inject on message
+    this.recursiveQueue = [];
+    this.recursiveNumber = 0;
+
     window.addEventListener('message', event => {
       // eslint-disable-next-line no-restricted-globals
-      if (event.data.nid === 'pageLoad' && event.data.data === location.href) {
-        injectAll();
+      if (event.data.nid === 'pageLoad') {
+        this.injectInpage().then(() => {
+          this.observer();
+        });
       }
     });
+  }
 
-    // inject on visibility
-    function handleVisibilityChange() {
-      // @ts-ignore
-      if (!document[hidden]) {
-        injectAll();
+  callback = (mutationsList: MutationRecord[]) => {
+    // Use traditional 'for loops' for IE 11
+    for (const mutation of mutationsList) {
+      if (mutation.type === 'childList') {
+        // @ts-ignore
+        const timeOut: number = setTimeout(() => {
+          this.injectInpage();
+        }, 1000);
+        this.recursiveQueue.push(timeOut);
       }
     }
-    if (typeof document.addEventListener === 'undefined' || hidden === undefined) {
-      console.log(
-        'This script requires a browser, such as Google Chrome or Firefox, that supports the Page Visibility API.'
-      );
+  };
+
+  observer() {
+    if (document.querySelector('.button-container')) {
+      const ele = document.querySelector('#meta-contents') as HTMLElement;
+      // @ts-ignore
+      if (ele?.children.length < 2) {
+        const timeOut = setTimeout(() => {
+          this.injectInpage();
+        }, 1000);
+        // @ts-ignore
+        this.recursiveQueue.push(timeOut);
+      }
+      const observer = new MutationObserver(this.callback);
+
+      const config = { attributes: true, childList: true, subtree: true };
+
+      observer.observe(ele, config);
     } else {
-      // Handle page visibility change
-      document.addEventListener(visibilityChange, handleVisibilityChange, false);
+      const timeOut = setTimeout(() => {
+        this.recursiveNumber += 1;
```

Comment:
Where is the `recursiveNumber ` used? Do we have a limit on max recursiveNumber ?

Reply:
Yes now have